examples/kqueue-process.rs

```diff
@@ -14,24 +14,36 @@
     target_os = "dragonfly",
 ))]
 fn main() -> std::io::Result<()> {
-    use std::process::Command;
+    use std::{num::NonZeroI32, process::Command};
 
     use async_io::os::kqueue::{Exit, Filter};
     use futures_lite::future;
 
     future::block_on(async {
         // Spawn a process.
-        let process = Command::new("sleep")
+        let mut process = Command::new("sleep")
             .arg("3")
             .spawn()
             .expect("failed to spawn process");
 
         // Wrap the process in an `Async` object that waits for it to exit.
-        let process = Filter::new(Exit::new(process))?;
+        let process_handle = unsafe {
+            Filter::new(Exit::from_pid(
+                NonZeroI32::new(process.id().try_into().expect("invalid process pid"))
+                    .expect("non zero pid"),
+            ))?
+        };
```

Comment:
I would prefer keeping the example simple.

Reply:
sgtm https://github.com/smol-rs/async-io/pull/207/commits/610773a62c852404514c797884696bc1f4201c91